javaServices/coreJavaServices/src/joliex/util/TimeService.java

```diff
@@ -415,4 +423,44 @@ public Long getTimestampFromString( Value request )
 			throw new FaultException( "InvalidTimestamp", pe );
 		}
 	}
+
+	@RequestResponse
+	public Long scheduleTimeout( Value request )
+	{
+		final long timeoutId = atomicLong.getAndIncrement();
+		TimeUnit unit;
+
+		if (request.hasChildren( "timeunit" )) {
+			try {
+				unit = TimeUnit.valueOf( request.getFirstChild( "timeunit" ).strValue().toUpperCase() );
+			} catch ( Exception e ) {
```

Comment:
This should probably throw an exception to the caller rather than succeeding silently.